app/models/comment.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
+
+  def increment_comments
+    user.increment!(:PostsCounter)
+  end
```

Comment:
- [ ] The Comment should belong to an `:author`, not `:user`. Kindly fix that.

  ```
  belongs_to :author, class_name: 'User'
  ```

- [ ] Kindly run the necessary migrations as well for the comments table to reference the `author`, instead `user`.

- [ ] We should increment the comments count of posts not, user. We should have

  ```
  def increment_comments
    post.increment!(:CommentsCounter)
  end
  ```

- **[OPTIONAL]** I think it would be better to implement a callback for this `increment_comments` method so that it is triggered whenever a post is saved.

  ```
  after _save :increment_comments
  
  private

  def increment_comments
    post.increment!(:CommentsCounter)
  end
  ```